docs/use-signalr-service.md

```diff
@@ -125,6 +125,24 @@ They can be accessed at [`Hub.Context.User`](https://github.com/aspnet/SignalR/b
     - When `Preferred`, Azure SignalR tries to find the app server which the client first negotiates with in a way that no additional cost or global routing is needed. This one can be useful when your scenario is *send to connection*. *Send to connection* can have better performance and lower latency when the sender and the receiver are routed to the same app server.
     - When `Required`, Azure SignalR always tries to find the app server which the client first negotiates with. This options can be useful when some client context is fetched from `negotiate` step and stored in memory, and then to be used inside `Hub`s. However, this option may have performance drawbacks because it requires Azure SignalR to take additional efforts to find this particular app server globally, and to keep globally routing traffics between client and server.
 
+
+#### `GracefulShutdown`
+
+##### Mode
+
+- Default value is `Off`
+- When set to `WaitForClientsClose`, instead of stopping the server immediately, we remove it from the ASRS to prevent new client connections from being assigned to this server.
+- When set to `MigrateClients`, in addition to the steps we mentioned in `WaitForClientsClose`, we will try migrate client connections to another valid server at a proper time. 
+  We could make sure the migration will only happen on the message boundaries, which means each of old/new servers will receive intact messages.
+  `OnConnected` and `OnDisconnected` will be triggered when connections be migrated in/out, and a `IConnectionMigrationFeature` will be set to help you determine if the connection has been migrated.
+  Please visit our sample codes for detail usage.
```

Comment:
also provide the link of the line of sample code using this IConnectionMigrationFeature?